### PR TITLE
Feature/initialize database with schema file

### PR DESCRIPTION
**Pull request by KhoiKy**
* The reserved migration IDs threshold has been changed from anything below `10000000000000` to anything below
`00010101000000`.
* `Initialize-Database` will now include the `schema.ps1` file from the database's migration directory if it exists.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
 
+# 0.17.0
+
+## Changes 
+
+* The reserved migration IDs threshold has been changed from anything below `10000000000000` to anything below
```
Comment (splatteredbits) [inline CHANGELOG.md:6]: Change these two bullets to: "* When initializing a database, Rivet now runs the migrations found in the schema.ps1 file, which contains the baseline database schema upon which all migrations should be applied. You can use the `Checkpoint-Migration` function to create a baseline `schema.ps1` file for your database(s)."

File: Rivet/Functions/Initialize-Database.ps1 [hunk 2]
```diff
@@ -9,13 +9,27 @@ function Initialize-Database
     param(
         [Parameter(Mandatory=$true)]
         [Rivet.Configuration.Configuration]
-        $Configuration
+        $Configuration,
+
+        [Switch] $Pop
```
Comment (splatteredbits) [inline Rivet/Functions/Initialize-Database.ps1:14]: Why did you add this `Pop` switch?

File: Rivet/Functions/New-Migration.ps1 [hunk 3]
```diff
@@ -38,6 +41,15 @@ function New-Migration
         $migrationPath = [IO.Path]::GetFullPath( $migrationPath )
         New-Item -Path $migrationPath -Force -ItemType File
 
+        $schemaPs1Path = Join-Path -Path $Path -ChildPath $script:schemaMigrationId
```
Comment (splatteredbits) [inline Rivet/Functions/New-Migration.ps1:44]: File name should be `$script:schemaFileName`.

File: Rivet/Functions/New-Migration.ps1 [hunk 4]
```diff
@@ -38,6 +41,15 @@ function New-Migration
         $migrationPath = [IO.Path]::GetFullPath( $migrationPath )
         New-Item -Path $migrationPath -Force -ItemType File
 
+        $schemaPs1Path = Join-Path -Path $Path -ChildPath $script:schemaMigrationId
+        if (-not (Test-Path -Path $schemaPs1Path))
+        {
+            $script:schemaPreamble | Set-Content -Path $schemaPs1Path
+            $msg = "Rivet created ""$($schemaPs1Path | Resolve-Path -Relative)"", a file where Rivet to stores a " +
```
Comment (splatteredbits) [inline Rivet/Functions/New-Migration.ps1:48]: Extra "to" in the warning message. Change "a" to "the", i.e. "stores the database's".

File: Rivet/Rivet.psm1 [hunk 5]
```diff
@@ -6,7 +6,22 @@ $RivetMigrationsTableName = 'Migrations'
 $RivetMigrationsTableFullName = "[$($RivetSchemaName)].[$($RivetMigrationsTableName)]"
 $RivetActivityTableName = 'Activity'
 $rivetModuleRoot = $PSScriptRoot
-[Int64] $script:firstMigrationId = 00010101000000
+$script:firstMigrationId = [Int64]'00010101000000' # 1/1/1 00:00:00
+$script:schemaMigrationId = [Int64]'00010000000000' # Special ID for schema.ps1, 1/0/0 00:00:00.
+$script:schemaFileName = 'schema.ps1'
+$script:schemaPreamble = @"
```
Comment (splatteredbits) [inline Rivet/Rivet.psm1:12]: Rename this variable to `$script:defaultSchemaPs1Content`.

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: Rivet/Functions/Initialize-Database.ps1 (hunk 2 above)
Comment (KhoiKy) [reply Rivet/Functions/Initialize-Database.ps1:14]: Invoke-Rivet function calls Initialize-Database, so when testing I noticed that when I ran Invoke-Rivet -Pop it would hit Initialize-Database and it wants to re-apply the schema.ps1 file but it has already been applied to the database so it throws an error. I also think we spoke about not popping the contents of the schema.ps1 file.

Comment (splatteredbits) [reply Rivet/Functions/Initialize-Database.ps1:14]: Hmm. Update-Database should be handling and skipping applying the schema.ps1 file if it's already been applied. Looks like that's a bug in Update-Database.
